controllers/cluster_controller.go

```diff
@@ -40,9 +40,10 @@ import (
 // ClusterReconciler reconciles a Cluster object
 type ClusterReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
-	DryRun bool
+	Log          logr.Logger
+	Scheme       *runtime.Scheme
+	DryRun       bool
+	CApiProvider string
```

Comment:
Can't we detect that ourselves in code? That would make things easier because no extra config is needed. For instance, we can use `provider.kind` from cluster-apps-operator config. Or alternatively: see that the `Cluster` object references an `AWSCluster` object – where we then only have to distinguish vintage AWS vs. CAPA if possible in a safe way.

Reply:
Definitely, nice idea @AndiDog; I looked into `provider.kind` and it's more consistent where `aws` = vintage, `capa` = AWS CAPA, `azure` = CAPA azure (don't know why it's not capz). For full provider list, which opsctl recognizes:
```
~> opsctl list installations --provider "test"
Error: Invalid flags: Provider must be empty, or one of: aws, azure, capa, cloud-director, common, gcp, kvm, openstack, vsphere
```

Reply:
I think one simple way of detecting whether or not a cluster is `vintage` is by checking the `release.giantswarm.io/version` label, which should be present only on `vintage` clusters. We don't use that concept in CAPI clusters.

Reply:
To know the specific provider, we can do what @AndiDog suggested and check the infraReference in the `Cluster` CR.

Reply:
`cluster-apps-operator` configmap works; tested for vintage (aws), azure, capa, and seems fine.

goku (capz):
```
1.684169771357345e+09   INFO    controllers.Cluster     Cluster org-df-test/2db3k provider is azure     {"cluster": "org-df-test/2db3k"}
```

akita (vintage):
```
1.684169752193825e+09   INFO    controllers.Cluster     Cluster org-adidas/pcn03 provider is aws        {"cluster": "org-adidas/pcn03"}
```

golem (capg):
```
1.684169735851858e+09   INFO    controllers.Cluster     Cluster org-marie/rp2y6 provider is capa        {"cluster": "org-marie/rp2y6"}
```